### PR TITLE
Add BIOS version and release date to SMBIOS

### PR DESCRIPTION
**Pull request by makubacki**
## Description

- Adds the BIOS version and date to Type 0 (BIOS) structure.
- Changes BIOS vendor to "Project Mu" so its easier to trace project info.
- Adds the QEMU version to the Type 1 (System) version field.
- Q35: Changes System product from "MuQemuQ35" to "QEMU Q35".
- SBSA: Changes System product from "MuQemuQ35" to "QEMU SBSA".
- SBSA: Fixes asset tag values from "Q35" to "SBSA"

Other related minor cleanup.

The goal is to clean up firmware and system information and make it more
useful.

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

1. Verified values in QEMU run command.
2. Read values from `smbiosview` in the EFI shell.
3. Read relevant values from Windows UI.

**BEFORE**

image

**AFTER**

image

## Integration Instructions

Review values modified and see if they affect tests or features
dependent on SMBIOS information.

### CODE REVIEW

**Review by os-d — APPROVED** [review]
LGTM. nit: In the PR description, you only mention changing Q35, you could also mention SBSA

**Comment by makubacki** [comment]
> LGTM. nit: In the PR description, you only mention changing Q35, you could also mention SBSA

Updated

File: Platforms/QemuQ35Pkg/Plugins/QemuRunner/QemuRunner.py [hunk 1]
```diff
@@ -175,8 +175,12 @@ def Runner(env):
...
 
-        args += " -smbios type=0,vendor=Palindrome,uefi=on"
-        args += " -smbios type=1,manufacturer=Palindrome,product=MuQemuQ35,serial=42-42-42-42,uuid=9de555c0-05d7-4aa1-84ab-bb511e3a8bef"
+        creation_time = Path(code_fd).stat().st_ctime
+        creation_datetime = datetime.datetime.fromtimestamp(creation_time)
+        creation_date = creation_datetime.strftime("%m/%d/%Y")
+
+        args += f" -smbios type=0,vendor=\"Project Mu\",version=\"mu_tiano_platforms-{repo_version}\",date={creation_date},uefi=on"
+        args += f" -smbios type=1,manufacturer=Palindrome,product=\"QEMU Q35\",family=QEMU,version=\"{'.'.join(qemu_version)}\",serial=42-42-42-42,uuid=9de555c0-05d7-4aa1-84ab-bb511e3a8bef"
         args += f" -smbios type=3,manufacturer=Palindrome,serial=40-41-42-43{boot_selection}"
```
Comment (TaylorBeebe) [inline Platforms/QemuQ35Pkg/Plugins/QemuRunner/QemuRunner.py:184]: Not sure why the serials are different here or what they're used for, but should they be made the same?

**Review by TaylorBeebe — APPROVED** [review]

File: Platforms/QemuQ35Pkg/Plugins/QemuRunner/QemuRunner.py (hunk 1 above)
Comment (makubacki) [reply Platforms/QemuQ35Pkg/Plugins/QemuRunner/QemuRunner.py:184]: I was looking at that too and ultimately decided to leave it alone. Mike Turner changed them a while back for DFCI testing - #589.

**Review by kuqin12 — APPROVED** [review]

**Pull request merged**